### DISCUSSION TITLE
Setting Icon for Raylib App on MacOS | Unix Executables/.app

### DISCUSSION THREAD

**Discussion by kennerhartman** [raysan5/raylib#3162] [Q&A] [answered]
By looking through the discussions on this repo, I know how to set an icon for a Windows executable.  

I tried using the *.rc.data file I generated for Windows on MacOS, but it does not set the icon for MacOS executables/.app (and I get a compiler error similar to the one down below).  How do I set the icon of a Raylib application for a MacOS executable/.app I generate?  I should note I am building this application in VS Code and not Xcode.

(I further tried using a *.plist, but I get a compiler error "ignoring file assets/Info.plist, building for macOS-x86_64 but attempting to link with file built for unknown-unsupported file format [hex code]".)

**Comment by raysan5** [comment, marked as answer]
I'm afraid macOS app icons work in a completely different way than Windows. The app is actually a zipped directory and icon must be a .icns file and placed in a very specific directory, I don't have the detailed info at hand now but you can check it on internet.

**Reply by kennerhartman** [reply]
I have figured it out (and it's a tedious process), but I do still have one more question.  I am trying to prevent Terminal from opening as I run the Raylib application.  On Windows I am able to supply "-mwindows" as I compile the program so it prevents the console from opening with the window.  Would you know how to achieve the same effect on macOS?

**Reply by raysan5** [reply]
Mmmmh... actually I don't know but I'd like to! I though `-mwindows` would work...

**Reply by kennerhartman** [reply]
I have tried this on macOS, but it doesn't work.  No compiler warnings are thrown, but it does not achieve the same effect as it would on Windows.

**Reply by kennerhartman** [reply]
I just tested this out for the App Bundle to stop the console from opening: 

Create a shell script that has the following code:
```
#!/bin/zsh
cd "${0%/*}"
open -a ./pong
```
and place it in pong.app/Contents/MacOS/

In /Contents/Info.plist, change the executable file from the unix executable to "pong.sh".  I guess what the ```-a``` is doing is suppressing the console.

Still, the problem is still present for the unix executable and app bundle unless I use "open -a".

**Reply by raysan5** [reply]
Thanks for the solution, that's weird... I think there should be some compilation option...

**Reply by kennerhartman** [reply]
What I said was not correct.

If the app is properly bundled, then there is no need for the script nor to use ```open -a```.  In other words, the terminal _will not_ show up when the Unix Executable is bundled as an app.  However, the terminal will still show up when running the Unix Executable, (even when compiled with ```-mwindows```) and instead you need to use ```open -a {executable file}``` to suppress the terminal.

I am closing this discussion.
